src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalsavoidnull/InputEqualsAvoidNullSuperClass.java

```diff
@@ -0,0 +1,26 @@
+/*
+EqualsAvoidNull
+ignoreEqualsIgnoreCase = (default)false
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.equalsavoidnull;
+
+public abstract class InputEqualsAvoidNullSuperClass {
+   protected String stringFromBaseClass = "ABC";
+}
+
+class DerivedClass extends InputEqualsAvoidNullSuperClass {
+   protected String classField = "DEF";
+
+    void m1() {
+        if (this.stringFromBaseClass.equals("JKHKJ")) { // ok
```

Comment:
this case is ok because this field is from superclass and we cannot get its type